core/src/main/java/hudson/slaves/SlaveComputer.java

```diff
@@ -992,6 +998,18 @@ public String call() throws IOException {
             catch (Throwable ex) { return "< 1.335"; } // Older agent.jar won't have VERSION
         }
     }
+
+    private static final class CommunicationProtocol extends MasterToSlaveCallable<String,IOException> {
+        @Override
+        public String call() throws IOException {
+            try {
+                Engine engine = Engine.current();
+                Method getProtocolName = Engine.class.getMethod("getProtocolName");
+                return (String) getProtocolName.invoke(engine);
```

Comment:
These three lines are same as
```java
return Engine.current().getProtocolName();
```
but `Engine.current().getProtocolName()` will be a compile error because `remoting.Engine` doesn't have `getProtocolName` method in the remoting version pointed in pom.xml.
So, after merging the PR in the remoting module, releasing remoting:4.8 and dump remoting version pointed in pom.xml to 4.8, We can change these lines to
```java
return Engine.current().getProtocolName();
```
I'm waiting for the feedback on this situation.

Reply:
Please add a TODO to detail this so the future reader will correct it when the time comes.